core/src/androidMain/kotlin/com/revenuecat/purchases/kmp/Purchases.android.kt

```diff
@@ -258,6 +256,14 @@ public actual class Purchases private constructor(private val androidPurchases:
         onSuccess = { onSuccess(it) },
     )
 
+    public actual fun recordPurchase(
+        productID: String,
+        onError: (error: PurchasesError) -> Unit,
+        onSuccess: (storeTransaction: StoreTransaction) -> Unit,
+    ): Unit = error(
+        "recordPurchase() is not supported on Android."
+    )
```

Comment:
I think that calling onError is definitely the better developer experience, since the existence of the `onError` parameter sort of implies that the function won't throw. I went ahead and updated this to call `onError` in commit https://github.com/RevenueCat/purchases-kmp/pull/139/commits/cda83262d8d2f1ed7657c97ecd76a131cdc81cd4.

It looks like we're currently calling `error()` in other places in the SDK when a function isn't supported, so this does introduce some inconsistency in how we handle unsupported functions. I'm okay shipping this either way (we can revert https://github.com/RevenueCat/purchases-kmp/pull/139/commits/cda83262d8d2f1ed7657c97ecd76a131cdc81cd4 if we don't want to keep it), but we should probably consider standardizing how we handle unsupported functions before the 1.0 release.

References:
- [Purchases.android.getPromotionalOffer](https://github.com/RevenueCat/purchases-kmp/blob/bdbadfe3d6deb3b0dacb06b502990c0b6de32687/core/src/androidMain/kotlin/com/revenuecat/purchases/kmp/Purchases.android.kt#L163) calls `error()` 
- [This Purchases.ios.purchase function](https://github.com/RevenueCat/purchases-kmp/blob/bdbadfe3d6deb3b0dacb06b502990c0b6de32687/core/src/iosMain/kotlin/com/revenuecat/purchases/kmp/Purchases.ios.kt#L210) calls `error`
- [Purchases.ios.syncAmazonPurchase](https://github.com/RevenueCat/purchases-kmp/blob/bdbadfe3d6deb3b0dacb06b502990c0b6de32687/core/src/iosMain/kotlin/com/revenuecat/purchases/kmp/Purchases.ios.kt#L126) is a no-op and doesn't call `error()` or throw

Reply:
Good call! I added [SDK-3554](https://linear.app/revenuecat/issue/SDK-3554/standardize-handling-of-functions-unsupported-on-a-single-platform).